src/main/java/com/fasterxml/jackson/databind/jsontype/impl/StdTypeResolverBuilder.java

```diff
@@ -398,4 +403,22 @@ protected boolean allowPrimitiveTypes(MapperConfig<?> config,
             JavaType baseType) {
         return false;
     }
+
+    /**
+     * Checks whether the given class has annotations indicating some type resolver
+     * is applied, for example {@link com.fasterxml.jackson.annotation.JsonSubTypes}.
+     * Only initializes {@link #_hasTypeResolver} once if its value is null.
+     *
+     * @param config the deserialization configuration to use
+     * @param baseType the base type to check for type resolver annotations
+     *
+     * @return true if the class has type resolver annotations, false otherwise
+     *
+     * @since 2.15
+     */
+    protected boolean _hasTypeResolver(DeserializationConfig config, JavaType baseType) {
+        AnnotatedClass ac = AnnotatedClassResolver.resolveWithoutSuperTypes(config,  baseType.getRawClass());
```

Comment:
Ahhhh. Only now realized WHY this actually works. Essentially this breaks introspection's inheritance of annotations, so indirectly checks, essentially, that target type has no `@JsonTypeInfo` annotation.

I guess that'll do for now; would be nice to have simpler checking (avoiding full construction of `TypeResolverBuilder` when all we need is existence check), but there are some downsides in adding new methods.

Reply:
Yes. And now that I started looking, I think I found a simple(r) fix for 3.0. Which is good... :)